File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/navdrawer/UserNavDrawerActivity.java
package com.jdroid.android.sample.ui.navdrawer;

import android.os.Bundle;

import com.jdroid.android.activity.FragmentContainerActivity;
import com.jdroid.android.auth.SecurityContext;
import com.jdroid.android.domain.User;
import com.jdroid.java.utils.RandomUtils;

import androidx.fragment.app.Fragment;

public class UserNavDrawerActivity extends FragmentContainerActivity {

	private User user;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		user = new User() {
			@Override
			public Long getId() {
				return RandomUtils.INSTANCE.getLong();
			}

			@Override
			public String getUserName() {
				return null;
			}

			@Override
			public String getFullname() {
				return "Tony Stark";
			}

			@Override
			public String getEmail() {
				return "tonystark@ironmail.com";
			}

			@Override
			public String getFirstName() {
				return null;
			}

			@Override
			public String getLastName() {
				return null;
			}

			@Override
			public String getUserToken() {
				return null;
			}

			@Override
			public String getProfilePictureUrl() {
				return null;
			}

			@Override
			public String getCoverPictureUrl() {
				return null;
			}
		};
		SecurityContext.INSTANCE.attach(user);
	}

	@Override
	protected void onStart() {
		super.onStart();

		SecurityContext.INSTANCE.attach(user);
	}

	@Override
	protected void onStop() {
		super.onStop();

		SecurityContext.INSTANCE.detachUser();
	}

	@Override
	protected Class<? extends Fragment> getFragmentClass() {
		return UserNavDrawerFragment.class;
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/navdrawer/UserNavDrawerActivity.kt
package com.jdroid.android.sample.ui.navdrawer

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.jdroid.android.activity.FragmentContainerActivity
import com.jdroid.android.auth.SecurityContext
import com.jdroid.android.domain.User
import com.jdroid.java.utils.RandomUtils

class UserNavDrawerActivity : FragmentContainerActivity() {

    private lateinit var user: User

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        user = object : User {
            override fun getId(): Long? {
                return RandomUtils.getLong()
            }

            override fun getUserName(): String? {
                return null
            }

            override fun getFullname(): String {
                return "Tony Stark"
            }

            override fun getEmail(): String {
                return "tonystark@ironmail.com"
            }

            override fun getFirstName(): String? {
                return null
            }

            override fun getLastName(): String? {
                return null
            }

            override fun getUserToken(): String? {
                return null
            }

            override fun getProfilePictureUrl(): String? {
                return null
            }

            override fun getCoverPictureUrl(): String? {
                return null
            }
        }
        SecurityContext.attach(user)
    }

    override fun onStart() {
        super.onStart()

        SecurityContext.attach(user)
    }

    override fun onStop() {
        super.onStop()

        SecurityContext.detachUser()
    }

    override fun getFragmentClass(): Class<out Fragment> {
        return UserNavDrawerFragment::class.java
    }
}
